orc8r/cloud/go/obsidian/swagger/handlers/handlers.go

```diff
@@ -33,3 +66,115 @@
+func GetSpecHandler(yamlCommon string) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		requestedService := c.Param("service")
+
+		service, ok, err := getServiceName(requestedService)
+		if err != nil {
+			return obsidian.HttpError(err, http.StatusInternalServerError)
+		}
+		if !ok {
+			return obsidian.HttpError(errors.New("service not found"), http.StatusNotFound)
+		}
+
+		combined, err := swagger.GetCombinedSpecFromService(yamlCommon, service)
+		if err != nil {
+			return obsidian.HttpError(err, http.StatusInternalServerError)
+		}
+
+		return c.String(http.StatusOK, combined)
+	}
+}
+
+// SpecInfo defines the URL to be injected into
+// the Swagger UI template.
+type SpecInfo struct {
+	URL string
+}
+
+// GetUIHandler returns a routing handler which serves
+// the UI of a service.
+func GetUIHandler(tmpl *template.Template, enableDynamicSwaggerSpecs bool) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		requestedService := c.Param("service")
+
+		service, ok, err := getServiceName(requestedService)
+		if err != nil {
+			return obsidian.HttpError(err, http.StatusInternalServerError)
+		}
+		if !ok {
+			return obsidian.HttpError(errors.New("service not found"), http.StatusNotFound)
+		}
+
+		// If runtime Swagger spec is off, serve static spec
+		if service == "" && !enableDynamicSwaggerSpecs {
+			service = "swagger.yml"
+		}
+
+		var buf bytes.Buffer
+		err = tmpl.Execute(&buf, SpecInfo{URL: service})
+		if err != nil {
+			return obsidian.HttpError(err, http.StatusInternalServerError)
+		}
+
+		return c.HTML(http.StatusOK, buf.String())
+	}
+}
+
+// registerSwaggerSpecHandlers registers routes for Swagger Specs.
+func registerSwaggerSpecHandlers(e *echo.Echo, trailSlashMiddleware echo.MiddlewareFunc) error {
+	yamlCommon, err := swagger.GetCommonSpec()
+	if err != nil {
+		return err
+	}
+
+	e.GET(obsidian.StaticURLPrefix+"/spec", nil, trailSlashMiddleware)
+	e.GET(obsidian.StaticURLPrefix+"/spec/", GetCombinedSpecHandler(yamlCommon))
+	e.GET(obsidian.StaticURLPrefix+"/spec/:service", GetSpecHandler(yamlCommon))
+
+	return nil
+}
+
+// registerSwaggerUIHandlers registers routes for the Swagger UI.
+func registerSwaggerUIHandlers(e *echo.Echo, trailSlashMiddleware echo.MiddlewareFunc) error {
+	tmpl, err := template.ParseFiles(obsidian.StaticFolder + "/swagger/v1/ui/index.html")
+	if err != nil {
+		return errors.Wrap(err, "retrieve Swagger template")
+	}
+
+	e.GET(obsidian.StaticURLPrefix+"/ui", nil, trailSlashMiddleware)
+	e.GET(obsidian.StaticURLPrefix+"/ui/", GetUIHandler(tmpl, obsidian.EnableDynamicSwaggerSpecs))
+	e.GET(obsidian.StaticURLPrefix+"/ui/:service", GetUIHandler(tmpl, obsidian.EnableDynamicSwaggerSpecs))
+
+	return nil
+}
+
+// getServiceName returns if the service is a registered
+// service with a Swagger spec.
+func getServiceName(service string) (string, bool, error) {
+	// If no service is provided, default to empty string to serve combined spec.
```

Comment:
Tiny nit/up to you: for inline comments (rather than doc comments), I prefer to not add a period unless it's serving as a full-stop before a new sentence